### PR TITLE
Feature/add file path parsing

### PR DESCRIPTION
**Pull request by ksobon**
### READY

This PR adds settings that will allow users to specify how the file path will be parsed to extract info from it about the Project. Currently HOK's network and folder structure is setup in a way that allows us to parse the project location, name and number from the file path. Using these generic settings and regex we can allow external users to define their own rules for extracting that info. It's used by the file path tools, when projects are published to MC. This then helps when searching file paths, and filtering the quite large list of file paths. 

image

Currently only FilePath is supported as source. I imagine in the future we can expand this to include things like Revit's Project Info object, and read info from parameters. 

We have options for each property to be pulled from a different source, as well as to specify different regex for different file path types ie. Revit Server, Local and BIM 360 since for example Project Name can be buried much deeper in the file path when we are dealing with Local file path as opposed to BIM 360.

### CODE REVIEW

**Comment by ksobon** [comment]
@djsiroky I added validations for all Regex inputs in here, so that we can at least point out to users when they are submitting a invalid Regex pattern. Please have a look and let me know.

**Comment by ksobon** [comment]
This is good. I Tried to keep the schema as simple as possible because I envision that we will have other sources of this information. One that I am trying to plan for is something like Project Info class in Revit. You can then specify what Parameter in Revit holds the Project Name, Project Number and Project Location info for any given file path. That would make the schema a little different where we wouldnt need to differentiate between bim360, local etc. It wouldn't matter. All that info would be in a Parameter. I will have a look at this, and see if it can be made a little easier to digest/less HTML code.

**Comment by ksobon** [comment]
@djsiroky I had a look at this and was able to shorten the code much using a small utility method, and a few `ng-repeats` without changing the schema that we have already. Have another look at it and let me know what you think.

File: public/angular-app/settings/settings.html [hunk 1]
```diff
@@ -273,488 +274,50 @@ <h3 class="panel-title">File Path Info</h3>
...
+                                                <i class="fas fa-asterisk fa-sm"
+                                                   style="color: red;"></i>
+                                            </button>
+                                        </label>
+                                    </div>
+                                    <div class="col-md-9 no-padding">
+                                        <div class="col-md-6 no-padding-left">
+                                            <input id="infoLocationPattern"
+                                                   type="text"
+                                                   placeholder="Location Regex"
```
Comment (djsiroky) [inline public/angular-app/settings/settings.html]: Is 'Location' in reference to Project Location / Project Name / Project Number or just a generic reference to filepath location? If the former, `placeholder="{{vm.processPropertyName(key)}} Regex"` should work here.

**Comment by ksobon** [comment]
@djsiroky can you have a final look at this and let me know if its good to go? Thanks!

**Pull request closed without merging**